ui/imports/shared/views/chat/LinksMessageView.qml

```diff
@@ -13,122 +13,81 @@ import shared.panels 1.0
-    property string links: ""
-    readonly property alias unfurledLinksCount: d.unfurledLinksCount
-    readonly property alias unfurledImagesCount: d.unfurledImagesCount
     property bool isCurrentUser: false
 
     signal imageClicked(var image, var mouse, var imageSource)
-    signal linksLoaded()
 
-    spacing: 4
+    spacing: 5
 
     Repeater {
         id: linksRepeater
-        model: linksModel
+        model: root.linkPreviewModel
+
         delegate: Loader {
             id: linkMessageLoader
-            required property var result
-            required property string link
-            required property int index
-            required property bool unfurl
-            required property bool success
-            required property bool isStatusDeepLink
-            readonly property bool isImage: result.contentType ? result.contentType.startsWith("image/") : false
-            readonly property bool neverAskAboutUnfurlingAgain: RootStore.neverAskAboutUnfurlingAgain
-            
-            active: success
+
+            // properties from the model
+            required property bool unfurled
+            required property string title
+            required property string description
+            required property string hostname
+            required property int thumbnailWidth
+            required property int thumbnailHeight
+            required property string thumbnailUrl
+            required property string thumbnailDataUri
+
             asynchronous: true
+            sourceComponent: unfurledLinkComponent
+
             StateGroup {
                 //Using StateGroup as a warkardound for https://bugreports.qt.io/browse/QTBUG-47796
-                id: linkPreviewLoaderState
-                states:[
+                states: [
                     State {
-                        name: "neverAskAboutUnfurling"
-                        when: !unfurl && neverAskAboutUnfurlingAgain
-                        PropertyChanges { target: linkMessageLoader; sourceComponent: undefined; }
-                        StateChangeScript { name: "removeFromModel"; script: linksModel.remove(index)}
-                    },
-                    State { 
-                        name: "askToEnableUnfurling"
-                        when: !unfurl && !neverAskAboutUnfurlingAgain
-                        PropertyChanges { target: linkMessageLoader; sourceComponent: enableLinkComponent }
-                    },
-                    State { 
-                        name: "loadImage"
-                        when: unfurl && isImage
-                        PropertyChanges { target: linkMessageLoader; sourceComponent: unfurledImageComponent }
-                    },
-                    State { 
                         name: "loadLinkPreview"
-                        when: unfurl && !isImage && !isStatusDeepLink
+                        when: !linkMessageLoader.isImage && !linkMessageLoader.isStatusDeepLink
```

Comment:
Missing properties
Do we plan to extend the model with these roles?

Reply:
https://github.com/status-im/status-desktop/issues/11634